packages/iota/test/validation/aliasOutput.spec.ts

```diff
@@ -0,0 +1,264 @@
+        const aliasOutput = cloneAliasOutput(mockAliasOutput);
+        if (aliasOutput.immutableFeatures) {
+            aliasOutput.immutableFeatures[1] = {
+                type: TAG_FEATURE_TYPE,
+                tag: "0xblablasometag"
+            };
+        }
+
+        const result = validateAliasOutput(aliasOutput, protocolInfoMock);
+
+        expect(result.isValid).toEqual(false);
+        expect(result.errors).toBeDefined();
+        expect(result.errors?.length).toEqual(1);
+        expect(result.errors).toEqual(expect.arrayContaining(
+            ["Alias output feature type of an Immutable Feature must define one of the following types: Issuer Feature or Metadata Feature."]
+        ));
+    });
+
+    it("should fail when the immutable features are not ordered in ascending order by type", () => {
+        const aliasOutput = cloneAliasOutput(mockAliasOutput);
+        aliasOutput.immutableFeatures = [
+            {
+                type: METADATA_FEATURE_TYPE,
+                data: "0xblablasomedata"
+            },
+            {
+                type: ISSUER_FEATURE_TYPE,
+                address: {
+                    type: ED25519_ADDRESS_TYPE,
+                    pubKeyHash: "0x6920b176f613ec7be59e68fc68f597eb3393af80f74c7c3db78198147d5f1f92"
+                }
+            }
+        ];
+
+        const result = validateAliasOutput(aliasOutput, protocolInfoMock);
+
+        expect(result.isValid).toEqual(false);
+        expect(result.errors).toBeDefined();
+        expect(result.errors?.length).toEqual(1);
+        expect(result.errors).toEqual(expect.arrayContaining(
+            ["Alias output Immutable Features must be sorted in ascending order based on their Immutable Feature Type."]
+        ));
+    });
+
+    it(
+        "should fail when the address in the state controller address unlock condition or Governor address unlock condition is the alias address itself (self-unlocking)",
+        () => {
+            const aliasOutput = cloneAliasOutput(mockAliasOutput);
+            aliasOutput.unlockConditions[0] = {
+                type: STATE_CONTROLLER_ADDRESS_UNLOCK_CONDITION_TYPE,
+                address: {
+                    type: ALIAS_ADDRESS_TYPE,
+                    aliasId: aliasOutput.aliasId
+                }
+            };
+
+            const result = validateAliasOutput(aliasOutput, protocolInfoMock);
+
+            expect(result.isValid).toEqual(false);
+            expect(result.errors).toBeDefined();
+            expect(result.errors?.length).toEqual(1);
+            expect(result.errors).toEqual(expect.arrayContaining(
+                ["Alias output Address field of the State Controller Address Unlock Condition and Governor Address Unlock Condition must not be the same as the Alias address derived from Alias ID."]
+            ));
+        }
+    );
+
+    it("should fail on when alias id is zeroed out and state index and foundry counter are not equal to zero", () => {
```

Comment:
Should fail when something "**but**" something else. Too much "and"-s in this sentence.